### PR TITLE
[feat](sync calendar): merge data team dashboard link in it

### PR DESCRIPTION
**Pull request by david30907d**
Added a metabase link placehlolder for now. Will replace it with the real `metabase link` once it's finished

### CODE REVIEW

File: sync_calendar.gs [hunk 1]
```diff
@@ -5,47 +5,17 @@ function onEdit(e) {
...
-        }
-      }
-    }
+  console.log("Sheet: " + sheet.getName());
+  if(sheet.getName() != "貼文表單") {
+    return;
   }
+
+  _updateScheduledPosts(range, sheet, row);
+  _insertDashboardLink(sheet, row, "報表(仮)");
```
Comment (david30907d) [inline sync_calendar.gs]: modularize the main onEdit function into these 2~

File: sync_calendar.gs [hunk 2]
```diff
@@ -265,4 +247,86 @@ function matchColor(status) {
...
+    sheet.getRange(1, 23).setValue("更新中");
+    reflashCalendar();
+    sheet.getRange(1, 23).setValue("--");
+  } else {
+    console.log("Place: "+ editRange[0].charCodeAt(0));
+
+    // 編輯 單一欄位，且欄位範圍符合條件則採取 刷新日曆
+    if(checkCellPlace(editRange[0].charCodeAt(0))) {
+      sheet.getRange(1, 23).setValue("更新中");
+      let data = getRowData(sheet, row);
```
Comment (david30907d) [inline sync_calendar.gs:307]: @winniepopu need to have your eyes on this~ Actually I didn't change anything but put them into this func

File: sync_calendar.gs [hunk 3]
```diff
@@ -117,7 +98,7 @@ function cleanCalender() {
   var sheet = SpreadsheetApp.getActive().getSheetByName('貼文日曆');
   let lastRow = sheet.getLastRow();
 
-  var range = sheet.getRange('D4:J'+lastRow);
+  var range = sheet.getRange('D27:J'+lastRow);
```
Comment (david30907d) [inline sync_calendar.gs:101]: other modifies are because of the inconsistences between github and AppScripts 😅

File: sync_calendar.gs [hunk 4]
```diff
@@ -103,8 +75,17 @@ function reflashCalendar(sheet) {
...
+      continue;
+    }
+    
     let tarIndex = addPostEvent(reqDatas[i].team, reqDatas[i].client, reqDatas[i].row, reqDatas[i].col, reqDatas[i].title, reqDatas[i].color);
     sheet.getRange(i+3, 5).setValue(tarIndex);
+
+    if(isTimeOut(startTime)) {
+      console.log("Time Out");
+      return;
+    }
```
Comment (david30907d) [inline sync_calendar.gs:88]: ditto

File: sync_calendar.gs (hunk 2 above)
Comment (gabriel0952) [reply sync_calendar.gs:307]: These changes are to fix the timeout issues during updates. Everything has been checked and looks good.

Thanks, @david30907d & @winniepopu

File: sync_calendar.gs (hunk 4 above)
Comment (gabriel0952) [reply sync_calendar.gs:88]: Same explanation as this [reply](https://github.com/pycontw/pycontw-post-calendar-tool/pull/3#discussion_r1623521926).

File: sync_calendar.gs (hunk 3 above)
Comment (gabriel0952) [reply sync_calendar.gs:101]: This is just to reduce changes to the completed calendar entries.

Comment (david30907d) [reply sync_calendar.gs:101]: got it~

**Review by winniepopu — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**